.github/workflows/upgrade-test.yml

```diff
@@ -123,7 +123,7 @@ jobs:
       - name: Prepare for Upgrade Testing
         run: |
           # Update the KfDef manifest with the candidate version
-          sed -i "s/main/${{ inputs.candidate-version }}/" ${{ env.RESOURCES_DIR }}/kfdef/kfdef.yaml
+          sed -i "s/${{ inputs.released-version }}/${{ inputs.candidate-version }}/" ${{ env.RESOURCES_DIR }}/kfdef/kfdef.yaml
```

Comment:
isn't this replacing the `main` here: https://github.com/opendatahub-io/data-science-pipelines-operator/blob/main/.github/resources/kfdef/kfdef.yaml#L16 

which is hard coded? why would it ever _not_ be `main` ?

Reply:
oh because you've already replaced it earlier, it would be better to just copy it to another working location